File: scratch/check_live_warmup.js
const DB_URL = "https://gestion-team-c-01-default-rtdb.firebaseio.com";

async function checkStatus() {
    try {
        const queueResp = await fetch(`${DB_URL}/state/autoWarmupQueue.json`);
        const queue = await queueResp.json() || {};
        console.log("=== CURRENT AUTO WARMUP QUEUE ===");
        console.log(JSON.stringify(queue, null, 2));

        const notifiedResp = await fetch(`${DB_URL}/state/autoWarmupNotified.json`);
        const notified = await notifiedResp.json() || {};
        console.log("\n=== RECENT NOTIFIED STATE (UPGRADES/DOWNGRADES) ===");
        console.log(JSON.stringify(notified, null, 2));

        const dataResp = await fetch(`${DB_URL}/warmupData.json`);
        const allData = await dataResp.json() || {};
        
        console.log("\n=== ACTIVE WARMUPS IN LAST 24 HOURS ===");
        const grouped = {};
        const cutoff = Date.now() - (24 * 60 * 60 * 1000);
        Object.values(allData).forEach(r => {
            if (!r.domain && !r.ip && !r.server) return;
            if (r.timestamp && r.timestamp < cutoff) return;
            const key = `${r.domain || ''}_${r.server || ''}_${r.ip || ''}`;
            if (!grouped[key]) grouped[key] = { ...r, records: [] };
            grouped[key].records.push(r);
        });

        for (const key in grouped) {
            const g = grouped[key];
            g.records.sort((a, b) => b.timestamp - a.timestamp);
            const latestDrop = g.records[0];
            const inVal = parseInt(latestDrop.inVal, 10) || 0;
            const outVal = parseInt(latestDrop.outVal, 10) || 0;
            console.log(`- Target: ${key}`);
            console.log(`  Latest drop: IN=${inVal}, OUT=${outVal}, Time=${new Date(latestDrop.timestamp).toLocaleString()}, Count=${g.records.length}`);
            
            // Check if last 3 succeeded
            let success = true;
            for (let i = 0; i < 3; i++) {
                if (g.records.length < 3) {
                    success = false;
                    break;
                }
                const r = g.records[i];
                const rIn = parseInt(r.inVal, 10) || 0;
                const rOut = parseInt(r.outVal, 10) || 0;
                if (rIn <= 0 || rOut < rIn * 0.95) {
                    success = false;
                    break;
                }
            }
            console.log(`  Succeed in 3 last drops: ${success}`);
        }
    } catch (e) {
        console.error(e);
    }
}

checkStatus();

File: scratch/check_live_warmup.py
import urllib.request
import json
import ssl
from datetime import datetime

DB_URL = "https://gestion-team-e-default-rtdb.firebaseio.com"
context = ssl._create_unverified_context()

def get_firebase(path):
    req = urllib.request.Request(f"{DB_URL}/{path}.json")
    try:
        with urllib.request.urlopen(req, context=context) as response:
            return json.loads(response.read().decode())
    except Exception as e:
        print(f"Error reading {path}: {e}")
        return None

print("=== CURRENT AUTO WARMUP QUEUE ===")
queue = get_firebase("state/autoWarmupQueue") or {}
print(json.dumps(queue, indent=2))

print("\n=== RECENT NOTIFIED STATE ===")
notified = get_firebase("state/autoWarmupNotified") or {}
print(json.dumps(notified, indent=2))

print("\n=== ACTIVE WARMUPS IN LAST 24 HOURS ===")
all_data = get_firebase("warmupData") or {}
grouped = {}
cutoff = datetime.now().timestamp() * 1000 - (24 * 60 * 60 * 1000)

for r in all_data.values():
    if not r.get("domain") and not r.get("ip") and not r.get("server"):
        continue
    t = r.get("timestamp", 0)
    if t < cutoff:
        continue
    key = f"{r.get('domain','')}_{r.get('server','')}_{r.get('ip','')}"
    if key not in grouped:
        grouped[key] = {**r, "records": []}
    grouped[key]["records"].append(r)

for key, g in grouped.items():
    records = sorted(g["records"], key=lambda x: x.get("timestamp", 0), reverse=True)
    latest = records[0]
    in_val = int(latest.get("inVal", 0))
    out_val = int(latest.get("outVal", 0))
    time_str = datetime.fromtimestamp(latest.get("timestamp", 0)/1000).strftime('%Y-%m-%d %H:%M:%S')
    print(f"- Target: {key}")
    print(f"  Latest drop: IN={in_val}, OUT={out_val}, Time={time_str}, Count={len(records)}")
    
    success = True
    if len(records) < 3:
        success = False
    else:
        for i in range(3):
            r = records[i]
            r_in = int(r.get("inVal", 0))
            r_out = int(r.get("outVal", 0))
            if r_in <= 0 or r_out < r_in * 0.95:
                success = False
                break
    print(f"  Succeed in 3 last drops: {success}")
